plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,87 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+  private File cue_file;
+  private File bin_file;
+	private string parent_directory;
+
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+
+	private size_t? header_offset;
+
+  public PlayStationHeader (File cue_file) {
+    this.cue_file = cue_file;
+  }
+
+  public void check_validity () throws PlayStationError {
+    check_cue_info ();
+
+		foreach (var offset in OFFSETS) {
+			var data = get_string_at_offset (offset, 50);
+			if (data.contains ("BOOT")) {
```

Comment:
It's considered a good practice to stop on invalid cases, like `if (!data.contains ("BOOT")) continue;`, if makes clear that we don't want to go further and avoid one level of indentation for the rest of the code.